src/app/roles.tsx

```diff
@@ -0,0 +1,28 @@
+"use client";
+
+import React, { useState, createContext } from "react";
+
+interface Props {
+    children: React.ReactNode;
+}
+
+interface RolesInterface {
+    [key: string]: string[];
```

Comment:
Just thinking because it is not taking general string but only "/clients" "/lists" "/calendar" and "/admin", would it worth making a type called pathname and only include these four, and share this type among all things that will use pathname? 
Similarly for role, maybe can make a type used globally that has only "admin" and "caller"?

Reply:
But I ain't sure if it is an overengineering (because it has to be updated whenever role and pathname has been created)
Let me know your thoughts